www/remotePackage.ts

```diff
@@ -59,41 +77,30 @@ class RemotePackage extends Package implements IRemotePackage {
                             successCallback && successCallback(localPackage);
                             Sdk.reportStatusDownload(localPackage, localPackage.deploymentKey);
                         });
-                    }, (fileError: FileError) => {
-                        CodePushUtil.invokeErrorCallback(new Error("Could not access local package. Error code: " + fileError.code), errorCallback);
-                    });
-                };
+                    }, fileError => onFileError(fileError, "READ_FILE"));
+                }
 
-                var downloadError = (error: FileTransferError) => {
-                    this.currentFileTransfer = null;
-                    CodePushUtil.invokeErrorCallback(new Error(error.body), errorCallback);
-                };
+                const filedir = cordova.file.dataDirectory + LocalPackage.DownloadDir + "/";
+                const filename = LocalPackage.PackageUpdateFileName;
 
-                this.currentFileTransfer.onprogress = (progressEvent: ProgressEvent) => {
-                    if (downloadProgress) {
-                        var dp: DownloadProgress = { receivedBytes: progressEvent.loaded, totalBytes: progressEvent.total };
-                        downloadProgress(dp);
-                    }
-                };
-
-                this.currentFileTransfer.download(this.downloadUrl, cordova.file.dataDirectory + LocalPackage.DownloadDir + "/" + LocalPackage.PackageUpdateFileName, downloadSuccess, downloadError, true);
+                cordova.plugin.http.downloadFile(this.downloadUrl, {}, {}, filedir + filename, onFileReady, onFileError);
             }
         } catch (e) {
             CodePushUtil.invokeErrorCallback(new Error("An error occured while downloading the package. " + (e && e.message) ? e.message : ""), errorCallback);
         }
     }
-    
+
     /**
      * Aborts the current download session, previously started with download().
-     * 
+     *
      * @param abortSuccess Optional callback invoked if the abort operation succeeded.
      * @param abortError Optional callback invoked in case of an error.
      */
     public abortDownload(abortSuccess?: SuccessCallback<void>, abortError?: ErrorCallback): void {
         try {
-            if (this.currentFileTransfer) {
-                this.currentFileTransfer.abort();
-            
+            if (this.isDownloading) {
+                this.isDownloading = false;
```

Comment:
This is questionable, but `cordova-plugin-advanced-http` does not implement `Canecelable` / has no abort triggers I could track down.